jobs/pacman-rule-engine-2.0/src/main/java/com/tmobile/pacman/util/CommonUtils.java

```diff
@@ -1265,5 +1266,42 @@ public static String eshttpGet(final String url, Map<String, String> headers) th
         }
     }
 
+    public static List<Map<String, String>> extractListOfMaps(String jsonString) throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        // Read the JSON string into a JsonNode
+        JsonNode rootNode = objectMapper.readTree(jsonString);
+
+        // Check if "message" is "success"
+        if (rootNode.has("message") && "success".equals(rootNode.get("message").asText())) {
+            // Extract "data" array
+            JsonNode dataArray = rootNode.get("data");
+
+            // Convert JsonNode to List<Map<String, Object>>
+            return convertJsonNodeToList(dataArray);
+        } else {
+            // Return an empty list if "message" is not "success"
+            return new ArrayList<>();
+        }
+    }
+
+    public static List<Map<String, String>> convertJsonNodeToList(JsonNode dataArray) {

```

Comment:
First, I do not see this method as reusable in the future. Would be better to have it as generic one
`public static List<T> convertJsonNodeToList(JsonNode dataArray)`
Second, its name is misleading because of its functionality - you converting not to list of some generic type, but exactly to list of maps.